src/CommitmentsV2/SFA.DAS.CommitmentsV2.UnitTests/Application/Commands/UpdateApprenticeshipStopDateCommandHandlerTests.cs

```
@@ -0,0 +1,387 @@
+            _dbContext = new ProviderCommitmentsDbContext(new DbContextOptionsBuilder<ProviderCommitmentsDbContext>()
+                                        .UseInMemoryDatabase(databaseGuid)
+                                        .ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.QueryClientEvaluationWarning))
+                                        .Options);
+
+            _confirmationDbContext = new ProviderCommitmentsDbContext(new DbContextOptionsBuilder<ProviderCommitmentsDbContext>()
+                            .UseInMemoryDatabase(databaseGuid)
+                            .ConfigureWarnings(warnings => warnings.Throw(RelationalEventId.QueryClientEvaluationWarning))
+                            .Options);
+
+            _currentDateTime = new Mock<ICurrentDateTime>();
+            _authenticationService = new Mock<IAuthenticationService>();
+            _nserviceBusContext = new Mock<IMessageSession>();
+            _encodingService = new Mock<IEncodingService>();
+            _overlapCheckService = new Mock<IOverlapCheckService>();
+            _overlapCheckService.Setup(x => x.CheckForOverlaps(It.IsAny<string>(), It.IsAny<CommitmentsV2.Domain.Entities.DateRange>(), It.IsAny<long?>(), It.IsAny<CancellationToken>()));
+            _logger = new Mock<ILogger<UpdateApprenticeshipStopDateCommandHandler>>();
+            _unitOfWorkContext = new UnitOfWorkContext();
+
+            _handler = new UpdateApprenticeshipStopDateCommandHandler(new Lazy<ProviderCommitmentsDbContext>(() => _dbContext),
+                _logger.Object,
+                _currentDateTime.Object,
+                _authenticationService.Object,
+                 _nserviceBusContext.Object,
+                _encodingService.Object,
+                _overlapCheckService.Object);
+                
+        }
+
+        [Test]
+        public async Task Handle_WhenHandlingCommand_WithInvalidData_ThenShouldThrowException()
```

Name of this test suggests it should throw an exception?

---

changed the method name